packages/tangle/scheduler.go

```diff
@@ -122,15 +126,32 @@ func (s *Scheduler) Setup() {
 			return
 		}
 
-		if err := s.SubmitAndReady(messageID); err != nil {
+		if err := s.Submit(messageID); err != nil {
 			if !errors.Is(err, schedulerutils.ErrBufferFull) &&
 				!errors.Is(err, schedulerutils.ErrInboxExceeded) &&
 				!errors.Is(err, schedulerutils.ErrInsufficientMana) {
 				s.Events.Error.Trigger(errors.Errorf("failed to submit to scheduler: %w", err))
 			}
 		}
+
+		s.updateDependencies(messageID, s.tryReady(messageID))
 	}))
 
+	onMessageScheduled := func(messageID MessageID) { s.updateDependencies(messageID, nil) }
+	s.tangle.Scheduler.Events.MessageScheduled.Attach(events.NewClosure(onMessageScheduled))
+
+	onMessageConfirmed := func(messageID MessageID) {
+		var scheduled bool
+		s.tangle.Storage.MessageMetadata(messageID).Consume(func(messageMetadata *MessageMetadata) {
+			scheduled = messageMetadata.Scheduled() || messageMetadata.ScheduledBypass() // TODO: remove bypass check
```

Comment:
A good rule on a PR that should be merged is to remove all TODOs

if this is supposed to be done as part of another issue maybe do the "ISSUE xxx:" comment

Reply:
Good point. The idea was to merge everything and then remove the scheduler bypassed flag to avoid potential conflicts if we did that before merging everything to `dev/drop-head`